tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java

```diff
@@ -61,11 +61,93 @@ public void testFunctionMocking() throws BallerinaTestException {
+            throw new BallerinaTestException(
+                    "Test failed due to default module single test failure.\nOutput:\n" + output);
+        }
+    }
+
+    @Test()
+    public void testObjectMocking_MockInvalidFieldName() throws BallerinaTestException {
+        String msg = "invalid field name 'invalidField' provided";
+        String[] args = mergeCoverageArgs(new String[]{"--tests", "object_mocking:testDefaultMockInvalidFieldName"});
+        String output =
+                balClient.runMainAndReadStdOut("test", args, new HashMap<>(),
+                        projectBasedTestsPath.resolve("object-mocking-tests").toString(), false);
+        if (!output.contains(msg)) {
+            throw new BallerinaTestException(
+                    "Test failed due to default module single test failure.\nOutput:\n" + output);
+        }
+    }
+
+    @Test()
+    public void testObjectMocking_MockInvalidReturnValue() throws BallerinaTestException {
+        String msg = "return value provided does not match the return type of function get()";
+        String[] args = mergeCoverageArgs(new String[]{"--tests", "object_mocking:testDefaultMockInvalidReturnValue"});
+        String output =
+                balClient.runMainAndReadStdOut("test", args, new HashMap<>(),
+                        projectBasedTestsPath.resolve("object-mocking-tests").toString(), false);
+        if (!output.contains(msg)) {
+            throw new BallerinaTestException(
+                    "Test failed due to default module single test failure.\nOutput:\n" + output);
+        }
+    }
+
+    @Test()
+    public void testObjectMocking_MockWrongAction() throws BallerinaTestException {
+        String msg = "return value provided does not match the return type of function get()";
+        String[] args = mergeCoverageArgs(new String[]{"--tests", "object_mocking:testDefaultMockWrongAction"});
+        String output =
+                balClient.runMainAndReadStdOut("test", args, new HashMap<>(),
+                        projectBasedTestsPath.resolve("object-mocking-tests").toString(), false);
+        if (!output.contains(msg)) {
+            throw new BallerinaTestException(
+                    "Test failed due to default module single test failure.\nOutput:\n" + output);
+        }
+    }
+
+    @Test()
+    public void testObjectMocking_TooManyArgs() throws BallerinaTestException {
+        String msg = "too many argument provided to mock the function get()";
+        String[] args = mergeCoverageArgs(new String[]{"--tests", "object_mocking:testDefaultTooManyArgs"});
+        String output =
+                balClient.runMainAndReadStdOut("test", args, new HashMap<>(),
+                        projectBasedTestsPath.resolve("object-mocking-tests").toString(), false);
+        if (!output.contains(msg)) {
+            throw new BallerinaTestException(
+                    "Test failed due to default module single test failure.\nOutput:\n" + output);
+        }
+    }
```

Comment:
Shall we write a single parameterized test using data providers for this test if possible?
Check `Passing Parameters with Dataproviders` part in [this tutorial](https://www.tutorialspoint.com/testng/testng_parameterized_test.htm).

Reply:
Condensed all the tests into one using data providers as suggested